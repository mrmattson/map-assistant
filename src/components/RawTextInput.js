import React from 'react'
import PropTypes from 'prop-types'
import {
  FormGroup,
  FormControl,
  ControlLabel,
  Button
} from 'react-bootstrap'

function RawTextInput ({ rawText, onRawTextChange, processRawText }) {
  return (
    <div className="RawTextInput">
      <FormGroup>
        <ControlLabel>Text Input</ControlLabel>
        <FormControl
          componentClass="textarea"
          className="RawTextInput-rawTextTextarea"
          onChange={event => onRawTextChange(event.target.value)}
          value={rawText}
        />
      </FormGroup>
      <Button
        bsSize="large"
        bsStyle="primary"
        onClick={() => processRawText(rawText)}
      >Process Text</Button>
    </div>
  )
}

// https://facebook.github.io/react/docs/typechecking-with-proptypes.html
RawTextInput.propTypes = {
  rawText: PropTypes.string.isRequired,
  onRawTextChange: PropTypes.func.isRequired,
  processRawText: PropTypes.func.isRequired
}

export default RawTextInput
