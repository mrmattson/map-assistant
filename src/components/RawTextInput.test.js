/* eslint-env jest */

import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import RawTextInput from './RawTextInput'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <RawTextInput
      rawText="rawText"
      onRawTextChange={function onRawTextChange () {}}
      processRawText={function processRawText () {}}
    />, div)
})

it('clean snapshot', () => {
  const component = renderer.create(
    <RawTextInput
      rawText="rawText"
      onRawTextChange={function onRawTextChange () {}}
      processRawText={function processRawText () {}}
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
