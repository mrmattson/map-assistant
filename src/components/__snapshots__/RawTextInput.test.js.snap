exports[`test clean snapshot 1`] = `
<div
  className="RawTextInput">
  <div
    className="form-group">
    <label
      className="control-label"
      htmlFor={undefined}>
      Text Input
    </label>
    <textarea
      className="RawTextInput-rawTextTextarea form-control"
      id={undefined}
      onChange={[Function]}
      type={undefined}
      value="rawText" />
  </div>
  <button
    className="btn btn-lg btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button">
    Process Text
  </button>
</div>
`;
